### PR TITLE
Update base rubocop config for new version, disabling most of the new things.

### PR DESCRIPTION
**Pull request by xaviershay**
I disabled all the new layout and style cops, but tried to keep the lint/perf/security ones.

Didn't bother with any kind of sorting/order, couldn't come up with a quick way to do and figured it wouldn't be that valuable.

### CODE REVIEW

File: travis/.rubocop_rspec_base.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # can customize by inheriting this file and overriding particular settings.
 
 AccessModifierIndentation:
-  EnforcedStyle: outdent
```
Comment (xaviershay) [inline travis/.rubocop_rspec_base.yml]: we weren't consistently respecting this in repos (particularly with `module_function`)

File: travis/.rubocop_rspec_base.yml [hunk 2]
```diff
@@ -47,9 +47,6 @@ DoubleNegation:
 EachWithObject:
   Enabled: false
 
-Encoding:
-  EnforcedStyle: when_needed
```
Comment (xaviershay) [inline travis/.rubocop_rspec_base.yml]: No longer respected by rubocop.

File: travis/.rubocop_rspec_base.yml [hunk 3]
```diff
@@ -95,9 +92,6 @@ PredicateName:
 Proc:
   Enabled: false
 
-RedundantReturn:
-  AllowMultipleReturnValues: true
```
Comment (xaviershay) [inline travis/.rubocop_rspec_base.yml]: Possibly could have figured out a way to keep this, but just disabled it instead.

**Comment by benoittgt** [comment]
I think we should merge this PR to try to be more synched to other repos.

**Comment by xaviershay** [comment]
I agree :)

**Comment by JonRowe** [comment]
@xaviershay Have you opened the PRs for the repos based on this?

**Comment by benoittgt** [comment]
- rspec-rails https://github.com/rspec/rspec-rails/pull/1962
- rspec-mocks https://github.com/rspec/rspec-mocks/pull/1206
- rspec-expectations (only ruby-2.6-preview1) https://github.com/rspec/rspec-expectations/pull/1050
- rspec-core is up-to-date https://github.com/benoittgt/rspec-core/commit/b8225ba888186b90138f0ad95f6a116b6f852dc1
- rspec-support (only ruby-2.6-preview1) https://github.com/rspec/rspec-support/pull/346

**Comment by xaviershay** [comment]
Yes, iirc they've been merged already. (Sorry traveling atm)


On Sat, Mar 3, 2018, at 11:54 PM, Jon Rowe wrote:
> @xaviershay[1] Have you opened the PRs for the repos based on this?> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].> 

Links:

  1. https://github.com/xaviershay
  2. https://github.com/rspec/rspec-dev/pull/198#issuecomment-370209277

**Comment by JonRowe** [comment]
Thanks @xaviershay wasn't sure if they had all been done to the same level, seems its just rspec-rails with differences.

**Pull request merged**